Plugin refactor [docs]: Rewrite README: update install, usage, workflows (#181)
DeepLabCut/napari-deeplabcut: README.md

[![codecov](https://codecov.io/gh/DeepLabCut/napari-deeplabcut/branch/main/graph/badge.svg)](https://codecov.io/gh/DeepLabCut/napari-deeplabcut)
[![napari hub](https://img.shields.io/endpoint?url=https://api.napari-hub.org/shields/napari-deeplabcut)](https://napari-hub.org/plugins/napari-deeplabcut)

A napari plugin for keypoint annotation and label refinement, also used within DeepLabCut.

---

## Installation

If you installed `DeepLabCut[gui]`, this plugin is already included.

You can also install `napari-deeplabcut` as a standalone keypoint annotation plugin without using the full DeepLabCut GUI.

### Standard install

Using `pip`:

```bash
pip install napari-deeplabcut
```

> Using `uv`:

Reviewer: Maybe we should make it a bit clear that this is not really an installation step, but adding it as project-managed dependency.

```suggestion
Using `uv` (adding it as project dependency):
```

Author: I changed it to pip install and added venv, that should be better. Thanks !

```bash
uv add napari-deeplabcut
```

> [!NOTE]
> A conda environment is not strictly required. Please use your preferred package manager!

### Latest development version

Using `pip`:

```bash
pip install git+https://github.com/DeepLabCut/napari-deeplabcut.git
```

---

## Usage

Start napari:

```bash
napari
```

Then activate the plugin in:

> **Plugins → napari-deeplabcut: Keypoint controls**

Accepted files such as `config.yaml`, image folders, videos, and `.h5` annotation files can be loaded either by dragging them onto the canvas or through the **File** menu.

> [!TIP]

> > The widget will open automatically when drag-and-dropping a compatible labeled data folder

Reviewer: ```suggestion
> The widget opens automatically when you drag-and-drop a compatible labeled-data folder.
```

- **Show trails**: display keypoint trails over time in the main viewer
- **Show trajectories**: open a trajectory plot in a separate dock widget
- **Show color scheme**: display the active/configured color mapping reference
- **Video tools**: extract the current frame and store crop coordinates for videos

---

## Saving layers

Use:

> **File → Save Selected Layer(s)...**

or the shortcut:

```text
Ctrl+S
```

### Keypoint save behavior

Keypoint annotations are automatically saved into the corresponding dataset folder as:

```text
CollectedData_<ScorerName>.h5
```

For convenience, the companion `.csv` file is written in the same folder.

### Important notes

- DeepLabCut uses the **H5** file as the authoritative annotation file.
- Before saving, make sure the **Points** layer you want to save is selected.
  - The plugin will not save if several Points layers are selected at the same time, to avoid ambiguity.
- Saving a `machinelabels...` layer does **not** write back to the machine labels file.
  Instead, refined annotations are written into the appropriate `CollectedData...` file.
- If saving would overwrite existing annotations, the plugin will prompt for confirmation.

>   - While labeling, confirmation can be disabled by unchecking the "Warn on overwrite" option in the dock widget.

Reviewer: These are an incredibly helpful addition. Makes things really clear for the user

Author: Thanks! Added a bit more re: config.yaml.
